src/app/homepage/homepage.component.ts

```diff
@@ -3,13 +3,46 @@ import { CommonModule } from '@angular/common';
 import { MatCardModule } from '@angular/material/card';
 import { RouterModule,RouterLink } from '@angular/router';
 
+import { UserSetCardComponent } from '../user-set-card/user-set-card.component';
+import { UserInfoFakeService } from '../services/user-info-fake.service';
+import { AccessData, UserData } from '../data-models/user-model';
+import { UserInfoService } from '../services/user-info.service';
+
 @Component({
   selector: 'app-homepage',
   standalone: true,
-  imports: [CommonModule,MatCardModule,RouterModule,RouterLink],
+  imports: [
+    CommonModule,
+    MatCardModule,
+    RouterModule,
+    RouterLink,
+    UserSetCardComponent
+  ],
   templateUrl: './homepage.component.html',
   styleUrl: './homepage.component.scss',
 })
 export class HomepageComponent {
+  // @Input() userData!:UserData;
+  // @Input() userId: string = "there wasn't a userId passed to this"; // remove later
+
+  userInfo: UserData = new UserData;
+  recentSetList: AccessData[] = this.userInfo.getRecentSets();
+  userSetList: AccessData[] = this.userInfo.getOwnedSets();
+
+  constructor(private userInfoFakeService: UserInfoFakeService) {};
```

Comment:
Could you change this to use the abstract UserInfoService. If it's not already added in app.config.ts you can add `{ provide: UserInfoService, useClass: UserInfoFakeService }` as a provider and then using it here will automatically use which ever service is defined there. You can change the name of the service to something like "userInfoService" as well.